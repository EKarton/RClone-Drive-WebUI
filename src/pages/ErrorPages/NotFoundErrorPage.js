import Button from '@mui/material/Button';
import axios from 'axios';
import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import BaseErrorPage from './BaseErrorPage';

export default function NotFoundErrorPage({ error, redirectText, redirectLink }) {
  const [imageUrl, setImageUrl] = useState();

  useEffect(() => {
    const fetchDogPictures = async () => {
      const { data } = await axios.get('https://dog.ceo/api/breeds/image/random');
      const pictureUrl = data.message;

      setImageUrl(pictureUrl);
    };

    fetchDogPictures();
  }, []);

  const renderButton = () => {
    return (
      <Button variant="outlined" component={Link} to={redirectLink}>
        {redirectText}
      </Button>
    );
  };

  return (
    <BaseErrorPage
      error={error}
      imageUrl={imageUrl}
      title="Sorry!"
      subtitle="This item might not exist or is no longer available"
      button={renderButton()}
    />
  );
}
