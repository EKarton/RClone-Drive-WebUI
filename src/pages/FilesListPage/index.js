import { FileViewerDialogProvider } from 'contexts/FileViewerDialog';
import { MoveFileDialogProvider } from 'contexts/MoveFileDialog';
import { RenameFileDialogProvider } from 'contexts/RenameFileDialog';
import useRemotePathParams from 'hooks/utils/useRemotePathParams';
import FilesListPageErrorBoundary from 'pages/ErrorBoundaries/FilesListPageErrorBoundary';
import Header from './Header';
import TableSection from './TableSection';
import './index.scss';

export default function FilesListPage() {
  const { remote, path } = useRemotePathParams();

  return (
    <FilesListPageErrorBoundary>
      <FileViewerDialogProvider>
        <MoveFileDialogProvider>
          <RenameFileDialogProvider>
            <div className="filelist-page__container">
              <Header remote={remote} path={path} />
              <TableSection remote={remote} path={path} />
            </div>
          </RenameFileDialogProvider>
        </MoveFileDialogProvider>
      </FileViewerDialogProvider>
    </FilesListPageErrorBoundary>
  );
}
