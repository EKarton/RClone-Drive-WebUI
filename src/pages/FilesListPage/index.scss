.filelist-page {
  &__container {
    display: flex;
    flex-direction: column;
    height: 100%;
    row-gap: 1rem;
  }
}
