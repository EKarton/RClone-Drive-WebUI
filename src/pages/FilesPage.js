import Toolbar from "@mui/material/Toolbar";
import "./FilesPage.scss";
import GlobalAppBar from "components/GlobalAppBar";
import FolderIcon from "@mui/icons-material/Folder";
import DescriptionIcon from "@mui/icons-material/Description";
import { Table } from "semantic-ui-react";

const FilesPage = () => {
  const rows = [
    { id: 1, col1: "Hello", col2: "World", col3: "123", col4: "Folder" },
    {
      id: 2,
      col1: "DataGridPro",
      col2: "is Awesome",
      col3: "343",
      col4: "File",
    },
    { id: 3, col1: "MUI", col2: "is Amazing", col3: "123", col4: "File" },
  ];

  const columns = [
    { field: "col1", headerName: "Name", flex: 1, minWidth: 150 },
    { field: "col2", headerName: "Last Updated", width: 200 },
    { field: "col3", headerName: "File Size", flex: 0.3, minWidth: 50 },
    { field: "col4", headerName: "Kind", flex: 0.3, minWidth: 50 },
  ];

  return (
    <>
      <GlobalAppBar />
      <Toolbar />
      <Table celled striped>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell colSpan="3">Git Repository</Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          <Table.Row>
            <Table.Cell collapsing>
              <FolderIcon /> node_modules
            </Table.Cell>
            <Table.Cell>Initial commit</Table.Cell>
            <Table.Cell collapsing textAlign="right">
              10 hours ago
            </Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell>
              <FolderIcon /> test
            </Table.Cell>
            <Table.Cell>Initial commit</Table.Cell>
            <Table.Cell textAlign="right">10 hours ago</Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell>
              <FolderIcon /> build
            </Table.Cell>
            <Table.Cell>Initial commit</Table.Cell>
            <Table.Cell textAlign="right">10 hours ago</Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell>
              <DescriptionIcon /> package.json
            </Table.Cell>
            <Table.Cell>Initial commit</Table.Cell>
            <Table.Cell textAlign="right">10 hours ago</Table.Cell>
          </Table.Row>
          <Table.Row>
            <Table.Cell>
              <DescriptionIcon /> Gruntfile.js
            </Table.Cell>
            <Table.Cell>Initial commit</Table.Cell>
            <Table.Cell textAlign="right">10 hours ago</Table.Cell>
          </Table.Row>
        </Table.Body>
      </Table>
    </>
  );
};

export default FilesPage;
