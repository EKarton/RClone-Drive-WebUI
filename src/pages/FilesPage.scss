.table {
  & .MuiDataGrid-cell:focus {
    outline: none !important;
  }
}
