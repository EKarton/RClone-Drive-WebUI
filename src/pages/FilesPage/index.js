import { useNavigate } from 'react-router-dom';
import RemotesListSection from 'components/RemotesListSection';
import { hashRemotePath } from 'utils/remote-paths-url';
import './index.scss';

export default function FilesPage() {
  const navigate = useNavigate();

  const handleRemoteCardClicked = (remote) => {
    navigate(`/files/${hashRemotePath(`${remote}:`)}`);
  };

  return (
    <div className="files-page">
      <RemotesListSection onRemoteCardClicked={handleRemoteCardClicked} />
    </div>
  );
}
