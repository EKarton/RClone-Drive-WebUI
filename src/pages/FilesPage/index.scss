.files-page {
  display: flex;
  flex-direction: column;
  row-gap: 1rem;
  padding-bottom: 3rem;
}
