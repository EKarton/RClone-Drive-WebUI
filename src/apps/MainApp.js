import Box from '@mui/material/Box';
import { Routes, Route } from 'react-router-dom';
import FilesListPageErrorBoundary from 'pages/ErrorBoundaries/FilesListPageErrorBoundary';
import PicturesListPageErrorBoundary from 'pages/ErrorBoundaries/PicturesListPageErrorBoundary';
import NotFoundErrorPage from 'pages/ErrorPages/NotFoundErrorPage';
import FilesListPage from 'pages/FilesListPage/index';
import FilesPage from 'pages/FilesPage/index';
import LandingPage from 'pages/LandingPage';
import LoginPage from 'pages/LoginPage';
import LogoutPage from 'pages/LogoutPage';
import PicturesListPage from 'pages/PicturesListPage/index';
import PicturesPage from 'pages/PicturesPage/index';
import AppShell from './AppShell';
import './MainApp.scss';

export default function MainApp() {
  return (
    <Box className="main-app" sx={{ bgcolor: 'background.default' }}>
      <Routes>
        <Route path="/" element={<LandingPage />} />
        <Route path="/login" element={<LoginPage />} />
        <Route path="files" element={<AppShell />}>
          <Route index element={<FilesPage />} />
          <Route
            index
            path=":id"
            element={
              <FilesListPageErrorBoundary>
                <FilesListPage />
              </FilesListPageErrorBoundary>
            }
          />
        </Route>
        <Route path="pictures" element={<AppShell />}>
          <Route index element={<PicturesPage />} />
          <Route
            index
            path=":id"
            element={
              <PicturesListPageErrorBoundary>
                <PicturesListPage />
              </PicturesListPageErrorBoundary>
            }
          />
        </Route>
        <Route path="/logout" element={<LogoutPage />} />
        <Route
          path="*"
          element={
            <NotFoundErrorPage
              error={new Error('Unknown path')}
              redirectLink="/"
              redirectText="Go back to home"
            />
          }
        />
      </Routes>
    </Box>
  );
}
