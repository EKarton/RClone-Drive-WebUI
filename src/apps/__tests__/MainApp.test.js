import MainApp from 'apps/MainApp';
import NotFoundErrorPage from 'pages/ErrorPages/NotFoundErrorPage';
import FilesListPage from 'pages/FilesListPage/index';
import FilesPage from 'pages/FilesPage/index';
import LandingPage from 'pages/LandingPage';
import LoginPage from 'pages/LoginPage';
import LogoutPage from 'pages/LogoutPage';
import PicturesListPage from 'pages/PicturesListPage/index';
import PicturesPage from 'pages/PicturesPage/index';
import { customRender } from 'test-utils/react';

jest.mock('pages/LandingPage');
jest.mock('pages/LoginPage');
jest.mock('pages/LogoutPage');
jest.mock('pages/FilesPage');
jest.mock('pages/FilesListPage');
jest.mock('pages/PicturesPage');
jest.mock('pages/PicturesListPage');
jest.mock('pages/ErrorPages/NotFoundErrorPage');

describe('MainApp', () => {
  it('should match snapshot given valid route', () => {
    LandingPage.mockReturnValue(null);

    const { baseElement } = customRender(<MainApp />, {}, { route: '/' });

    expect(baseElement).toMatchSnapshot();
  });

  it.each([
    [LandingPage, '/'],
    [LoginPage, '/login'],
    [LogoutPage, '/logout'],
    [FilesPage, '/files'],
    [FilesListPage, '/files/123'],
    [PicturesPage, '/pictures'],
    [PicturesListPage, '/pictures/123'],
    [NotFoundErrorPage, '/random-path'],
  ])('should render %s given route is %s', (expectedComponent, route) => {
    expectedComponent.mockReturnValue(null);

    customRender(<MainApp />, {}, { route });

    expect(expectedComponent).toBeCalled();
  });
});
