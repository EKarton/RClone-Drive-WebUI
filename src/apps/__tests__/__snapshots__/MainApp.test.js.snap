// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainApp should match snapshot given valid route 1`] = `
<body>
  <div>
    <div
      class="main-app MuiBox-root css-1k9ek97"
    />
  </div>
</body>
`;
